### PR TITLE
Allow changes only to collaborative/my own articles

### PR DESCRIPTION
**Pull request by dkocich**
Ready to review
Close https://github.com/c2corg/v6_ui/issues/1029

### CODE REVIEW

**Comment by tsauerwein** [comment]
Could you please add tests for this?

File: c2corg_api/views/article.py [hunk 1]
```diff
@@ -57,6 +62,23 @@ def collection_post(self):
...
+                    self.request.validated['document']['article_type']
+                if article_type != article.article_type:
+                    raise HTTPBadRequest(
+                        'Personal article cannot be changed'
+                    )
+            # personal articles should only be modifiable by
+            # their creator and moderators
+            elif not has_been_created_by(article_id,
+                                         self.request.authenticated_userid):
+                raise HTTPForbidden('No permission to change this image')
```
Comment (tsauerwein) [inline c2corg_api/views/article.py]: image?

File: c2corg_api/views/article.py [hunk 2]
```diff
@@ -57,6 +62,24 @@ def collection_post(self):
                 validate_article_update,
                 validate_associations_update])
     def put(self):
+        if not self.request.has_permission('moderator'):
+            article_id = self.request.validated['id']
+            article = DBSession.query(Article).get(article_id)
+            if article is None:
+                raise HTTPNotFound('No article found for id %d' % article_id)
```
Comment (tsauerwein) [inline c2corg_api/views/article.py]: Elsewhere we are using `'... {}'.format(article_id)`.

Comment (dkocich) [inline c2corg_api/views/article.py]: Ok, sometimes yes, sometimes no - like [here for image](https://github.com/c2corg/v6_api/blob/master/c2corg_api/views/image.py#L144). I will change both now but there are still around +-50 occurrences of "...' % ..." elsewhere. I am not sure if they all can be just simply changed to `.format` instead

**Pull request merged**